middleware/compress.go

```diff
@@ -38,7 +38,8 @@ type Compressor struct {
 	// The mapping of pooled encoders to pools.
 	pooledEncoders map[string]*sync.Pool
 	// The set of content types allowed to be compressed.
-	allowedTypes map[string]bool
+	allowedTypes     map[string]struct{}
+	allowedWildcards map[string]struct{}
```

Comment:
This might be a contentious change.

I always prefer `struct{}` when a map is used to check only for existence, because then I'm never left wondering "does the bool value mean something?"

It also uses less memory (one byte less memory per map entry), which is, at best, a minor concern here, where maps probably contain tens of items, or less, on average.

I've met a few fellow developers who hate this convention, though.  So if this bothers anyone, I can revert to `bool` :)

Reply:
since its an internal type and `struct{}` takes less memory in the map, better to go with the `struct{}` if its just intended to work as a marker